src/data/option/basic.lean

```diff
@@ -105,6 +111,31 @@ by cases x; simp
 
 @[simp] theorem map_id' : option.map (@id α) = id := map_id
 
+/--
+Composing an `option.map` with another `option.map` is equal to
+a single `option.map` of a composition of functions, fully applied.
+This is the reverse direction of `option.comp_map`.
+-/
+@[simp] lemma map_map (h : β → γ) (g : α → β) (x : option α) :
+  option.map h (option.map g x) = option.map (h ∘ g) x :=
+by { cases x; simp only [map_none', map_some'] }
+
+/--
+A single `option.map` of a composition of functions is equal to
+composing an `option.map` with another `option.map`, fully applied.
+This is the reverse direction of `option.map_map`.
+-/
+lemma comp_map (h : β → γ) (g : α → β) (x : option α) :
+  option.map (h ∘ g) x = option.map h (option.map g x) := (map_map _ _ _).symm
+
+/--
+Composing an `option.map` with another `option.map` is equal to
+a single `option.map` of composed functions.
```

Comment:
I don't think this comment tells us anything that the lemma statement doesn't already tell us.

Reply:
Happy to remove any docstrings. A general question -- should new lemmas added to mathlib have docstrings added? It's, of course, easier to contribute them without docstrings. Which ones _do_ require docstrings? Should the docstring remark on its similarity to the functor equivalent, but more general due to universes?

Reply:
In my opinion: 
- Lemmas should definitely have docstrings if their statement is not clear from a quick glance at the type.
- Otherwise, docstrings are encouraged if they provide helpful information. This is subjective, of course. There may be cases where restating the type in English can help and other cases where it makes things more obscure. 
- Here's a non-exhaustive list of other info that's good to have in docstrings: 
  - pointers to related lemmas or definitions in mathlib that might be hard to find just from the statement or the naming convention,
  - usage notes,
  - conventional names of important results, 
  - citations to the literature

Reply:
@cipher1024 do you mind making some concrete suggestions for the doc strings so we can proceed (i.e. do you prefer that this is removed or edited somehow)?

Reply:
I like your list @bryangingechen. Some of those uses can go in module doc strings, you can have a heading introducing a number of lemmas and describe different things about them. I would add that a lemma doc string can also be useful when the lemma breaks a guideline knowingly and provide an explanation. `[note]` is also useful for that when it's a common pattern but I don't have intimate information about `[note]`.

Reply:
Thanks! Yes, I might PR this list to our guidelines at some point. However, the main thrust of my question was to find out what you wanted for the doc strings in *this* PR, since it seemed like that was one of the last unresolved points.

Reply:
I see. I was simply suggesting to delete the comment.

Reply:
OK, docstrings removed